core/include/keyman/keyman_core_api.h

```diff
@@ -1140,6 +1140,58 @@ KMN_API
 km_core_status
 km_core_state_get_intermediate_context(km_core_state *state, km_core_context_item ** context_items);
 
+/*
+```
+### Context Debug Reporting
+As of version 17, the cached context is an internal property of the
+state, not exposed to the consumer of the API -- apart from the
+Keyman Developer Keyboard Debugger. However, for other debug
+purposes, it is helpful to be able to examine the cached context, so
+a debug-formatted version of the context is made available with
+`km_core_state_context_debug`. This is not intended to be parsed for
+reading the context for other purposes, and the format may change.
+
+```c
+*/
+typedef enum {
+  KM_CORE_DEBUG_CONTEXT_CACHED        = 0,
+  KM_CORE_DEBUG_CONTEXT_INTERMEDIATE  = 1
+} km_core_debug_context_type;
+
+/*
+```
+### `km_core_state_context_debug`
+##### Description:
+Returns a debug formatted string of the context from the state.
+##### Return:
+A pointer to a km_core_cp UTF-16 string. Must be disposed of by a call
+to `km_core_cp_dispose`.
+##### Parameters:
+- __state__: A pointer to the opaque state object to be queried.
+- __context_type__: the type of context to retrieve from the state
```

Comment:
A few more words explaining this parameter might be useful. It's not immediately obvious to me what `KM_CORE_DEBUG_CONTEXT_INTERMEDIATE` will return.

Reply:
More context is added in https://github.com/keymanapp/keyman/pull/10382